client/components/AddSponsor.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+
+interface AddSponsorProps {
+  eventId: string;
+  chapterId: string;
+}
+
+interface SponsorData {
+  name?: string;
+  website?: string;
+  type?: string;
+}
+
+const AddSponsor: React.FC<AddSponsorProps> = ({ eventId, chapterId }) => {
+  const [isSubmitting, setSubmitting] = React.useState<boolean>(false);
```

Comment:
No need to specify types if they can be implied from the initial value, so this and `success` can just be `React.useState(false)`

Reply:
changed. Though if we are allowing implicit types why use ts at all? (curiosity question)